### ISSUE TITLE
GM_XHR with responseType document passes charset to DOMParser that cause error

### ISSUE THREAD

**Issue by 7nik** [Tampermonkey/tampermonkey#1000]
GM_xmlhttpRequest passes the value of the Content-Type header to DOMParser without dropping charset and the parser throws error about unsupported mime-type.

## Expected Behavior
The script prints to console a Document object.

## Actual Behavior
Warning in the console:
> GM_xmlhttpRequest: TypeError: Failed to execute 'parseFromString' on 'DOMParser': The provided value 'text/html; charset=utf-8' is not a valid enum value of type SupportedType.

## Specifications

Chromium: Version 83.0.4103.97 (Official Build) Arch Linux (64-bit)
TM: beta, 4.11.6116
OS: KDE Manjaro (64-bit)

## Script

```js
// ==UserScript==
// @name         Test GM_XHR responseType
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @grant        GM_xmlhttpRequest
// @connect      httpbin.org
// ==/UserScript==

(() => {
    GM_xmlhttpRequest({
        url: "https://httpbin.org/html",
        responseType: "document",
        onload: ({ response }) => console.log(response),
    });
})();
```

**Comment by 7nik** [comment]
The same for option `fetch: true`.

**Comment by derjanb** [comment]
Should be fixed at 4.11.6117

**Comment by dodying** [comment]
regression at v4.12
